dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionLogMapper.xml

```diff
@@ -50,6 +50,16 @@
         where code = #{code}
         and version = #{version}
     </select>
+    <select id="queryByDefinitionCodesAndVersion" resultType="org.apache.dolphinscheduler.dao.entity.ProcessDefinitionLog"
+            parameterType="org.apache.dolphinscheduler.dao.entity.ProcessInstance">
+        select
+        <include refid="baseSql"/>
+        from t_ds_process_definition_log
+        where
+        <foreach collection="processInstances" item="item" separator="or">
```

Comment:
It should avoid the use `or` in query, because it will miss the index and query slowly.

Reply:
Ok, I took this change advice from wenjun. Let me check the sql explain. If it doesn't use index in reality, we have to query `ProcessDefinitionLog` one by one here.